docs/reference/transform/transforms-at-scale.asciidoc

```diff
@@ -0,0 +1,197 @@
+
+If you have defined a {transform} source index `query`, ensure it is as 
+efficient as possible. Use the **Search Profiler** under **Dev Tools** in {kib} 
+to get detailed timing information about the execution of individual components 
+in the search request. Alternatively, you can use the <<search-profile>>. The 
+results give you insight into how search requests are executed at a low level so 
+that you can understand why certain requests are slow, and take steps to improve 
+them.
+
+{transforms-cap} execute standard {es} search requests. There are different ways 
+to write {es} queries, and some of them are more efficient than others. Consult 
+<<tune-for-search-speed>> to learn more about {es} performance tuning.
+
+
+[discrete]
+[[index-sorting-group-by-ordering]]
+== 3. Use Index Sorting and `group_by` ordering (search, process)
+
+If you use more than one `group_by` field in your {transform}, then the order of 
+the fields in conjunction with the use of <<index-modules-index-sorting>> may 
+improve runtime.
+
+Index sorting enables you to store documents on disk in a specific order which 
+can improve query efficiency. The ideal sorting logic depends on your use case, 
+but the rule of thumb may be to sort the fields in descending order (high to low 
+cardinality) starting with the time-based fields. Then put the time-based 
+components first in the `group_by` if you have any, and then apply the same 
+order to your `group_by` fields as configured for index sorting. 
+
+
+[discrete]
+[[increase-shards-dest-index]]
+== 4. Increase the number of shards of the destination index (index)
+
+<<scalability,Depending on the size of the destination index>>, you may consider 
+increasing its shard count. {transforms-cap} use one shard by default when 
+creating the destination index. To override the index settings, create the 
+destination index before starting the {transform}.
+
+Hint: Use the <<preview-transform>> to check the settings that the {transform} 
+would use to create the destination index. You can copy and adjust these in 
+order to create the destination index prior to starting the {transform}.
+
+
+[discrete]
+[[optimize-shading-strategy]]
+== 5. Optimize the sharding strategy for the source index (search)
+
+There is no one-size-fits-all sharding strategy. A strategy that works in one 
+environment may not scale in another. A good sharding strategy must account for 
+your infrastructure, use case, and performance expectations.
+
+Too few shards may mean that the benefits of distributing the workload cannot be 
+realised; however too many shards may impact your cluster health. To learn more 
+about sizing your shards, read this <<size-your-shards,guide>>.
+
+
+[discrete]
+[[disable-source-dest]]
+== 6. Disable the `_source` field on the destination index (storage)
+
+The <<mapping-source-field>> contains the original JSON document body that was 
+passed at index time. The `_source` field itself is not indexed (and thus is not 
```

Comment:
> that was passed ...

This wording makes me wonder: "Passed to where?" or "Passed to what?". Is there a clearer way to state this?